src/visGeometry/index.ts

```diff
@@ -1136,54 +1213,39 @@ class VisGeometry {
         this.idColorMapping.clear();
     }
 
-    private getColorIndexForTypeId(typeId: number): number {
-        const index = this.idColorMapping.get(typeId);
-        if (index === undefined) {
-            this.logger.error(
-                "getColorIndexForTypeId could not find " + typeId
-            );
-            return 0;
-        }
-        return index % (this.colorsData.length / 4);
-    }
-
-    private getColorForTypeId(typeId: number): Color {
-        const index = this.getColorIndexForTypeId(typeId);
-        return this.getColorForIndex(index);
-    }
-
-    private setColorForId(id: number, colorId: number): void {
-        /**
-         * @param id agent id
-         * @param colorId index into the color array
-         */
-        this.idColorMapping.set(id, colorId);
-
-        // if we don't have a mesh for this, add a sphere instance to mesh registry?
-        if (!this.visGeomMap.has(id)) {
-            this.visGeomMap.set(id, DEFAULT_MESH_NAME);
-        }
-    }
-
     public setColorForIds(ids: number[], colorId: number): void {
         /**
          * Sets one color for a set of ids, using an index into a color array
          * @param ids agent ids that should all have the same color
-         * @param colorId index into the color array
+         * @param colorId index into the numberOfColors
          */
-        ids.forEach((id) => this.setColorForId(id, colorId));
+        ids.forEach((id) => {
+            this.setColorForId(id, colorId);
+        });
     }
 
     public applyColorToAgents(agentIds: number[], colorId: number): void {
+        /**
+         * Sets one color for a set of ids
+         */
         this.setColorForIds(agentIds, colorId);
         this.updateScene(this.currentSceneAgents);
     }
 
-    public getColorForIndex(index: number): Color {
+    public getColorForColorId(colorId: number): Color {
+        if (colorId < 0) {
```

Comment:
I think these cases will be impossible to hit. I wrote them as I was thinking of edge cases writing tests.